internal/engine/target_queue.go

```diff
@@ -7,37 +7,113 @@ import (
 	"github.com/windmilleng/tilt/internal/store"
 )
 
-// A little data structure to help iterate through targets in dependency order.
+// Allows the caller to inject its own build strategy for dirty targets.
```

Comment:
fwiw @maiamcc i think an interface that looks like this one will eventually replace BuildAndDeployer...

we'll define a bunch of functions that take
1) a single target spec
2) the current build state of that target
3) the results of the dependencies

we'll have different handlers for building them, and a central build graph (like TargetQueue) that calls the build handlers whenever a target is marked dirty